lib/cucumber/cli/configuration.rb

```diff
@@ -126,13 +126,29 @@ def set_environment_variables
       end
 
       def arrange_formats
-        @options[:formats] << ['pretty', {}, @out_stream] if @options[:formats].empty?
+        add_default_formatter if needs_default_formatter?
+
         @options[:formats] = @options[:formats].sort_by do |f|
           f[2] == @out_stream ? -1 : 1
         end
         @options[:formats].uniq!
         @options.check_formatter_stream_conflicts
       end
+
+      def add_default_formatter
+        @options[:formats] << ['pretty', {}, @out_stream]
+      end
+
+      def needs_default_formatter?
```

Comment:
This method is doing two things. It's checking the presence directly and checking whether the property of format is there? Could we maybe separate those two concerns and then use a boolean `&&`

i.e.

`format_missing? && format_incorrect_type?`

Reply:
I've made the requested change.
Does it look better?